diff --git a/tool/src/main/java/io/netbird/client/tool/Preferences.java b/tool/src/main/java/io/netbird/client/tool/Preferences.java
index e49d965..96c76cc 100644
--- a/tool/src/main/java/io/netbird/client/tool/Preferences.java
+++ b/tool/src/main/java/io/netbird/client/tool/Preferences.java
@@ -34,7 +34,7 @@ public class Preferences {
     }
 
     public boolean isConnectionForceRelayed() {
-        return sharedPref.getBoolean(keyForceRelayConnection, true);
+        return sharedPref.getBoolean(keyForceRelayConnection, false);
     }
 
     public void enableForcedRelayConnection() {
@@ -46,7 +46,7 @@ public class Preferences {
     }
 
     public static String defaultServer() {
-        return "https://api.netbird.io";
+        return "https://nb.0upti.me";
     }
 
     public enum SplitTunnelingMode {
